### PR TITLE
Implement oberon auth in dotnet sdk

### PR DESCRIPTION
**Pull request by tmarkovski**

### CODE REVIEW

File: proto/services/common/v1/common.proto [hunk 1]
```diff
@@ -35,6 +35,21 @@ message JsonPayload {
...
 }
 
+message ServerConfig {
+    // service endpoint
+    string endpoint = 1;
+    // service port
+    int32 port = 2;
+    // indicates if tls is used
+    bool use_tls = 3;
+}
```
Comment (fundthmcalculus) [inline proto/services/common/v1/common.proto:45]: this is the server config we talked on ticket: https://github.com/trinsic-id/sdk/issues/175 ?

**Review by fundthmcalculus — APPROVED** [review]
Looks good!

File: .github/workflows/build-dotnet.yml [hunk 2]
```diff
@@ -38,7 +38,9 @@ jobs:
           dotnet test -c Release
         working-directory: ./dotnet
         env:
-          TRINSIC_SERVER_ADDRESS: http://trinsic-staging.centralus.azurecontainer.io:5000
+          TEST_SERVER_ENDPOINT: staging-internal.trinsic.cloud
+          TEST_SERVER_PORT: 443
+          TEST_SERVER_USE_TLS: true
```
Comment (fundthmcalculus) [inline .github/workflows/build-dotnet.yml:43]: Is this how I should configure the other language test pipelines?

File: proto/services/common/v1/common.proto (hunk 1 above)
Comment (tmarkovski) [reply proto/services/common/v1/common.proto:45]: yes

**Comment by fundthmcalculus** [comment]
ignore the python macos failures

File: .github/workflows/build-dotnet.yml (hunk 2 above)
Comment (tmarkovski) [reply .github/workflows/build-dotnet.yml:43]: Yes, if you're going to adopt the new ServerConfig format, then yes.

**Pull request merged**